pages/api/levels/query.ts

```diff
@@ -0,0 +1,102 @@
+import withAuth, { NextApiRequestWithAuth } from '../../../lib/withAuth';
+
+import Level from '../../../models/db/level';
+import { LevelModel } from '../../../models/mongoose';
+import type { NextApiResponse } from 'next';
+import dbConnect from '../../../lib/dbConnect';
+
+export default withAuth(async (req: NextApiRequestWithAuth, res: NextApiResponse) => {
+  if (req.method !== 'GET') {
+    return res.status(405).json({
+      error: 'Method not allowed',
+    });
+  }
+
+  await dbConnect();
+
+  const { name, username, author_note, min_moves, max_moves, time_range, sort_by, sort_dir } = req.query as {name:string, username: string, author_note:string, min_moves:string, max_moves:string, time_range:string, min_rating:string, sort_by:string, sort_dir:string};
+  const searchObj = {} as {[key:string]:any};
+  const limit = 10;
+
+  if (name) {
+    searchObj['name'] = {
+      $regex: name,
+      $options: 'i',
+    };
+  }
+
+  if (author_note) {
+    searchObj['authorNote'] = {
+      $regex: author_note,
+      $options: 'i',
+    };
+  }
+
+  if (min_moves && max_moves) {
```

Comment:
Could we allow using only min or max as well? Might want to search for only short levels < 100 moves for example

Reply:
i would think though we could just pass in 0, max and min,1e9 or something of the sort to handle that and just change the ux